src/plugins/intel_gpu/src/graph/impls/onednn/utils.cpp

```diff
@@ -350,6 +368,36 @@ static dnnl::memory::format_tag get_format_by_desc(dnnl::memory::desc desc) {
     return dnnl::memory::format_tag::undef;
 }
 
+cldnn::format find_data_format(dnnl::memory::desc desc) {
+    auto onednn_desc = get_format_by_desc(desc);
+
+    if (onednn_desc != dnnl::memory::format_tag::undef) {
+        return convert_data_format(onednn_desc);
+    } else {
+        auto blk = desc.data.format_desc.blocking;
+        if (desc.data.ndims == 5 && blk.inner_nblks == 1
```

Comment:
Please consider implementing more generic code here. I think we can try to match memory::desc with our format traits (i.e. compare format rank, blocking desc and dims order). This may allow us to get rid of code like below and some maps between cldnn and onednn tags.

Reply:
Sure